lib/rules/indent.js

```diff
@@ -348,6 +348,22 @@ module.exports = {
         }
 
         /**
+         * Returns the ExpressionStatement based on the current node
+         * if not present then return null
+         * @param {ASTNode} node node to examine
+         * @returns {ASTNode|void} if found then node otherwise null
+         */
+        function getAssignmentExpressionNode(node) {
```

Comment:
Rather than duplication the whole  functionality of this function, i would recommend doing something like this.

```js
const getParentNodeByType = (node, type) => {
    let parent = node.parent;

    while (parent.type !== type && parent.type !== "Program") {
        parent = parent.parent;
    }

    return parent.type === type ? parent : null;
}

const getVariableDeclaratorNode = (node) => getParentNodeByType(node, "VariableDeclarator");

const getAssignmentExpressionNode = (node) => getParentNodeByType(node, "AssignmentExpression");
```